src/link.ts

```diff
@@ -51,11 +51,17 @@ export type Link = {
 
 }
 
+type NewLink = Omit<Link, 'context'>;
```

Comment:
Could you describe what is happening here? A new link is omitting a Link type or context string?

Reply:
We're creating a new type that removes the `context` property from `Link`. Below, we allow either `NewLink` or `Link`. The real effective goal here was to make `context` optional for callers.

Reply:
Interesting, what are the differences between this approach and simply rewriting Link to have the context property as optional?

Reply:
If `context` is optional on the Link type it means anything that uses the Link type will now always need to check if context is defined or not. In reality context is only optional when adding it to the collection of links, any consumer of this object will guaranteed get a `context`.